cpp/src/arrow/meson.build

```diff
@@ -459,24 +463,27 @@ if needs_ipc
     }
 endif
 
-if needs_tests
-    foreach key, val : arrow_tests
-        exc = executable(
-            key,
-            sources: val['sources'],
-            dependencies: [arrow_test_dep, val.get('dependencies', [])],
-        )
-        test(key, exc)
-    endforeach
-endif
+foreach key, val : arrow_tests
+    exc = executable(
+        key,
+        sources: val['sources'],
+        dependencies: [arrow_test_dep, val.get('dependencies', [])],
+    )
+    test(key, exc)
+endforeach
 
 if needs_benchmarks
     benchmark_dep = dependency(
         'benchmark',
         default_options: {'tests': 'disabled'},
     )
+
+    arrow_benchmark_dep = declare_dependency(
+        link_with: [arrow_test_lib],
+        dependencies: [arrow_dep, gtest_main_dep, benchmark_dep],
```

Comment:
Can we use `benchmark_main` instead of `gtest_main` for benchmark?

Reply:
Yes we can. The only possible downside to doing so is that we would still need the gtest dependency, just without the main executable attached to it